test/plugins/BufferCreator.cpp

```diff
@@ -0,0 +1,185 @@
+/**
+ * @file BufferCreator.cpp BufferCreator class
+ * implementation
+ *
+ * This is part of the DUNE DAQ Software Suite, copyright 2020.
+ * Licensing/copyright details are in the COPYING file that you should have
+ * received with this code.
+ */
+
+//#include "CommonIssues.hpp"
+#include "BufferCreator.hpp"
+#include "trigger/Issues.hpp"
+#include "trigger/buffercreator/Nljs.hpp"
+
+#include "appfwk/DAQModuleHelper.hpp"
+#include "appfwk/app/Nljs.hpp"
+
+#include "logging/Logging.hpp"
+
+#include <chrono>
+#include <cstdlib>
+#include <string>
+#include <thread>
+#include <vector>
+
+namespace dunedaq {
+namespace trigger {
+
+BufferCreator::BufferCreator(const std::string& name)
+  : dunedaq::appfwk::DAQModule(name)
+  , m_thread(std::bind(&BufferCreator::do_work, this, std::placeholders::_1))
+  , m_queueTimeout(100)
+  , m_input_queue_tps()
+  , m_input_queue_dr()
+  , m_output_queue_frag()
+  , m_buffer()
+{
+  register_command("conf", &BufferCreator::do_configure);
+  register_command("start", &BufferCreator::do_start);
+  register_command("stop", &BufferCreator::do_stop);
+  register_command("scrap", &BufferCreator::do_scrap);
+}
+
+void
+BufferCreator::init(const nlohmann::json& init_data)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering init() method";
+
+  m_input_queue_tps.reset(new appfwk::DAQSource<trigger::TPSet>(appfwk::queue_inst(init_data, "tpset_source")));
```

Comment:
May as well use the `*_source_t` and `*_sink_t` aliases you defined in the header file here